rust-bins/src/bin/keygen.rs

```diff
@@ -259,15 +269,23 @@ fn handle_generate_ar_keys(kgar: KeygenAr) -> Result<(), String> {
         }
 
         if !kgar.no_confirmation {
-            // clear screen
-            execute!(std::io::stdout(), Clear(ClearType::All))
-                .map_err(|_| "Could not clear screen.".to_owned())?;
-            println!("Please enter recovery phrase again to confirm.");
-
-            let confirmation_words = read_words_from_terminal(kgar.in_len, true, &bip39_map)?;
-
-            if confirmation_words != randomized_words {
-                return Err("Recovery phrases do not match.".to_string());
+            let mut first = true;
```

Comment:
Variable could be avoided by printing initial prompt before loop and only print retry request if check fails at the end.